packages/x-scheduler-headless/src/process-event/processEvent.ts

```diff
@@ -1,18 +1,40 @@
-import { SchedulerEvent, SchedulerProcessedEvent } from '../models';
+import { SchedulerEvent, SchedulerProcessedEvent, SchedulerValidDate } from '../models';
 import { processDate } from '../process-date';
 import { Adapter } from '../use-adapter';
 import { parseRRuleString } from '../utils/recurring-events';
+import { TemporalTimezone } from '../base-ui-copy/types';
+
+export function processEvent(
+  model: SchedulerEvent,
+  timezone: TemporalTimezone,
+  adapter: Adapter,
+): SchedulerProcessedEvent {
+  const startTimezone = adapter.getTimezone(model.start);
+  const endTimezone = adapter.getTimezone(model.end);
+
+  if (startTimezone && endTimezone && startTimezone !== endTimezone) {
```

Comment:
```suggestion
  if (startTimezone !== endTimezone) {
```

Both variable are always defined